--- a/esphome/writer.py
+++ b/esphome/writer.py
@@ -404,7 +404,7 @@ def get_build_info() -> tuple[int, int, str, str]:
         Tuple of (config_hash, build_time, build_time_str, comment)
     """
     config_hash = CORE.config_hash
-    build_time = int(time.time())
+    build_time = int(os.environ["SOURCE_DATE_EPOCH"])
     build_time_str = time.strftime("%Y-%m-%d %H:%M:%S %z", time.localtime(build_time))
     comment = CORE.comment or ""
     return config_hash, build_time, build_time_str, comment
